src/deploy/deploy.py

```diff
@@ -262,6 +263,46 @@ def deploy(ctx, release_id, environment_id, description):
     )
 
 
+@cli.command()
+@click.option('--release-id', prompt="Release ID to deploy", default="latest", show_default=True,
+              help="The ID of the release to be deployed, or the latest release if unspecified")
+@click.option('--environment-id', prompt="Environment ID to deploy release to",
+              default="stage", show_default=True,
+              help="The target environment of this deployment")
+@click.pass_context
+def confirm_deploy(ctx, release_id, environment_id, wait_for_seconds = 600, interval = 10):
+    start_timer = time.perf_counter()
+    project = ctx.obj['project']
+    release = project.get_release(release_id)
+
+    def _confirm_deploy():
+        is_release_deployed = project.is_release_deployed(release, environment_id)
+        waited_for = time.perf_counter() - start_timer
+        
+        if not is_release_deployed and waited_for < wait_for_seconds:
+            click.echo(click.style(f"Deployment of {release_id} to {environment_id} is not complete. Waiting {interval} seconds", fg="yellow"))
+            time.sleep(interval)
+            _confirm_deploy()
+
+        if not is_release_deployed and waited_for >= wait_for_seconds:
+            return False
+        
+        if is_release_deployed:
+            return True
+
+    click.echo("")
+    click.echo(click.style(f"Checking deployment of {release_id} to {environment_id}", fg="yellow"))
+
+    click.echo("")
+
+    confirmed = _confirm_deploy()
+    if confirmed:
+        click.echo(click.style(f"Deployment of {release_id} to {environment_id} successful", fg="bright_green"))
+        return 0
+    else:
+        click.echo(click.style(f"Deployment of {release_id} to {environment_id} failed", fg="yellow"))
```

Comment:
nit: do we know it failed, or just that it didn't successfully complete within the time?

Reply:
For now it's not confirmed within a timespan.
I might have a think if we can multiple states to report on e.g. `undeployed` <= but this might be quite fiddly to work out.